### PR TITLE
chore(TPGOV-2071): Implement get sample metadata endpoint V2

### PR DESCRIPTION
**Pull request by bjacobtalend**
**What is the problem this Pull Request is trying to solve?**
 
**What is the chosen solution to this problem?**
 
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TPGOV-2071
 
**Please check if the Pull Request fulfills these requirements**
- [ ] The PR commit message follows our [guidelines](https://github.com/Talend/daikon/blob/master/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

 
**[ ] This Pull Request introduces a breaking change**

### CODE REVIEW

File: daikon-schema/daikon-dataset-sample/README.md
Comment (fabballe) [inline daikon-schema/daikon-dataset-sample/README.md:1]: wrong copy paste ;)

File: daikon-schema/daikon-dataset-sample/pom.xml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            <artifactId>junit-jupiter-engine</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-params</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
+        </dependency>
```
Comment (fabballe) [inline daikon-schema/daikon-dataset-sample/pom.xml:34]: as daikon is moved to java 17 now. You may be able to remove lombok dep

File: daikon-schema/daikon-dataset-sample/src/main/java/org/talend/daikon/schema/dataset/sample/metadata/DqRuleQuality.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+/**
+ * Pojo representation of {@link sample/dqRuleQuality.json}
+ */
+@Value
+@SuperBuilder(toBuilder = true)
+@Jacksonized
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class DqRuleQuality {
```
Comment (fabballe) [inline daikon-schema/daikon-dataset-sample/src/main/java/org/talend/daikon/schema/dataset/sample/metadata/DqRuleQuality.java:18]: as daikon is moved to java 17 now. You may be able to replace it by a record

File: daikon-schema/daikon-dataset-sample/src/main/java/org/talend/daikon/schema/dataset/sample/metadata/RecordField.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+/**
+ * Pojo representation of {@link sample/recordField.json}
+ */
+@Value
+@SuperBuilder(toBuilder = true)
+@Jacksonized
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class RecordField {
```
Comment (fabballe) [inline daikon-schema/daikon-dataset-sample/src/main/java/org/talend/daikon/schema/dataset/sample/metadata/RecordField.java:20]: as daikon is moved to java 17 now. You may be able to replace it by a record

File: daikon-schema/daikon-dataset-sample/src/main/java/org/talend/daikon/schema/dataset/sample/metadata/RecordFieldQuality.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+/**
+ * Pojo representation of {@link sample/recordFieldQuality.json}
+ */
+@Value
+@SuperBuilder(toBuilder = true)
+@Jacksonized
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class RecordFieldQuality {
```
Comment (fabballe) [inline daikon-schema/daikon-dataset-sample/src/main/java/org/talend/daikon/schema/dataset/sample/metadata/RecordFieldQuality.java:20]: as daikon is moved to java 17 now. You may be able to replace it by a record

File: daikon-schema/daikon-dataset-sample/src/main/java/org/talend/daikon/schema/dataset/sample/metadata/SampleMetadata.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+/**
+ * Pojo representation of {@link sample/sampleMetadata.json}
+ */
+@Value
+@Builder(toBuilder = true)
+@Jacksonized
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class SampleMetadata {
```
Comment (fabballe) [inline daikon-schema/daikon-dataset-sample/src/main/java/org/talend/daikon/schema/dataset/sample/metadata/SampleMetadata.java:20]: as daikon is moved to java 17 now. You may be able to replace it by a record

File: daikon-schema/daikon-dataset-sample/src/test/resources/sampleMetadata/sampleMetadataInvalid.json
Comment (fabballe) [inline daikon-schema/daikon-dataset-sample/src/test/resources/sampleMetadata/sampleMetadataInvalid.json:1]: From a Data Plane/Control Plane point of view you must not mix data (value) with metadata (quality) on the same schema/endpoint

**Review by fabballe — CHANGES_REQUESTED** [review]

File: daikon-schema/daikon-dataset-sample/src/main/java/org/talend/daikon/schema/dataset/sample/metadata/DqRuleQuality.java (hunk 2 above)
Comment (bjacobtalend) [reply daikon-schema/daikon-dataset-sample/src/main/java/org/talend/daikon/schema/dataset/sample/metadata/DqRuleQuality.java]: replaced by record

File: daikon-schema/daikon-dataset-sample/src/main/java/org/talend/daikon/schema/dataset/sample/metadata/RecordField.java (hunk 3 above)
Comment (bjacobtalend) [reply daikon-schema/daikon-dataset-sample/src/main/java/org/talend/daikon/schema/dataset/sample/metadata/RecordField.java]: replaced by record

File: daikon-schema/daikon-dataset-sample/src/main/java/org/talend/daikon/schema/dataset/sample/metadata/SampleMetadata.java (hunk 5 above)
Comment (bjacobtalend) [reply daikon-schema/daikon-dataset-sample/src/main/java/org/talend/daikon/schema/dataset/sample/metadata/SampleMetadata.java]: replaced by record

File: daikon-schema/daikon-dataset-sample/src/test/resources/sampleMetadata/sampleMetadataInvalid.json
Comment (bjacobtalend) [reply daikon-schema/daikon-dataset-sample/src/test/resources/sampleMetadata/sampleMetadataInvalid.json:1]: value removed

**Review by fabballe — CHANGES_REQUESTED** [review]
You have some tests on deserialization but no test on serialization. You should add some

File: daikon-schema/daikon-dataset-sample/src/main/resources/sample/dqRuleQuality.json [hunk 6]
```diff
@@ -0,0 +1,20 @@
+{
+  "$schema": "https://json-schema.org/draft/2019-09/schema",
+  "$id": "https://org.talend.daikon/dqRule.quality.schema.json",
+  "title": "Dq Rule quality schema",
+  "description": "Dq Rule quality on Talend ecosystem",
+  "type": "object",
+  "properties": {
+    "id": {
+      "description": "rule uuid",
```
Comment (fabballe) [inline daikon-schema/daikon-dataset-sample/src/main/resources/sample/dqRuleQuality.json:9]: if it is an UUID you should add `format: "uuid"`

Comment (bjacobtalend) [reply daikon-schema/daikon-dataset-sample/src/main/resources/sample/dqRuleQuality.json:9]: done

**Pull request merged**